src/sentry/relay/config/metric_extraction.py

```diff
@@ -240,16 +236,33 @@ def _get_widget_metric_specs(
                 return []
 
     max_widget_specs = options.get("on_demand.max_widget_specs") or _MAX_ON_DEMAND_WIDGETS
-    if len(specs) > max_widget_specs:
-        sentry_sdk.set_tag("organization_id", project.organization_id)
+    specs = _trim_if_above_limit(specs, max_widget_specs, project, "widgets")
+
+    return specs
+
+
+def _trim_if_above_limit(
+    specs: Sequence[HashedMetricSpec],
+    max_specs: int,
+    project: Project,
+    widget_type: str,
+) -> list[HashedMetricSpec]:
+    """If we have more specs than the max limit we should trim it."""
+    return_specs = list(specs)
+
+    if len(specs) > max_specs:
         # Do not log for Sentry
         if project.organization.id != 1:
```

Comment:
It would be better to use an option for this since it's independent of the Sentry's instance. If a user uses their own sentry instance or we rework our entity ids, this code will misbehave.

Reply:
I added it to my to-do.